database/rotator.go

```diff
@@ -239,29 +264,25 @@ func rotatePartitions(db *dbr.Connection, query string, d, p int) {
 	t := time.Now().Add(time.Hour * time.Duration(24*d))
 	oldName := "pnr0000"
 	newName := "pnr0"
+	OriQuery := query
 
 	startTime := time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, t.Location())
 	oldStart := "StartTime"
-	newStart := startTime.Add(time.Minute * time.Duration(0)).Format("2006-01-02 15:04:05-07:00")
 
 	endTime := time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, t.Location())
 	oldEnd := "EndTime"
-	newEnd := endTime.Add(time.Minute * time.Duration(p)).Format("2006-01-02 15:04:05-07:00")
 
-	for i := 0; i < 1440/p; i++ {
-		if i > 0 {
-			newName = "pnr" + strconv.Itoa(i)
-			newStart = startTime.Add(time.Minute * time.Duration(i*p)).Format("2006-01-02 15:04:05-07:00")
-			newEnd = endTime.Add(time.Minute * time.Duration(i*p+p)).Format("2006-01-02 15:04:05-07:00")
-		}
+	for i := 1; i < 1440/p; i++ {
+		query := OriQuery
+		newName = "pnr" + strconv.Itoa(i)
+		newStart := startTime.Add(time.Minute * time.Duration(i*p)).Format("2006-01-02 15:04:05-07:00")
+		newEnd := endTime.Add(time.Minute * time.Duration(i*p+p)).Format("2006-01-02 15:04:05-07:00")
+
 		query = strings.Replace(query, oldName, newName, -1)
-		oldName = newName
 
 		query = strings.Replace(query, oldStart, newStart, -1)
-		oldStart = newStart
 
-		query = strings.Replace(query, oldEnd+"');", newEnd+"');", -1)
-		oldEnd = newEnd
+		query = strings.Replace(query, oldEnd, newEnd, -1)
```

Comment:
This will break postgres. As you see might see simple string.Replace is used. For postgres you have following simplified ranges:

00.00-01:00
01:00-02:00

Now you can see that 01:00 would be replaced twice. Thats why I was looking at the proper end
strings.Replace(query, oldEnd+"');", newEnd+"');", -1)

Reply:
This works for mysql because we have no overlapping there. Only
less 01:00
less 02.00